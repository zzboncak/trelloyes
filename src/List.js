import React from 'react';
import './List.css';
import Card from "./Card";

class List extends React.Component {
    constructor(props){
        super(props);
    }
    
    static defaultProps = {
        cards: [],
    };
    
    render() {
        return (
            <section className="List">
            <header className="List-header">
                <h2>{this.props.header}</h2>
            </header>
            <div className="List-cards">
                {this.props.cards.map(card => (
                    <Card 
                        key = {card.id}
                        id = {card.id}
                        listId = {this.props.id}
                        title = {card.title}
                        content = {card.content}
                        handleDelete = {this.props.handleDelete}
                    />
                ))}
                <button type="button" className="List-add-button">
                    + Add Random Card
                </button>
            </div>
        </section>
        )
    }
}


export default List;
