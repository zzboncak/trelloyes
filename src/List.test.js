import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe("List component", () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<List />)
          .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})