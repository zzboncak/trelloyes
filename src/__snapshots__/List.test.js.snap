// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
